src/actions/articleAction.js

```diff
@@ -0,0 +1,35 @@
+import { ACTIONS } from './actionTypes';
+import axios from 'axios';
+
+/**
+ * @description - articles response function
+ * @param {string} actionType - type of action
+ * @param {object} articles - articles returned
+ * @returns {object}
+ */
+export const articleAction = (actionTypes, articles) => ({
+  type: actionTypes,
+  payload: articles
+});
+
+/**
+ * @description Request to the API to get only 12 latest published articles
+ * @returns {object} dispatch object
+ */
+
+export const getArticles = () => async(dispatch) => {
+  try {
+    dispatch(articleAction(ACTIONS.LOADING, 'started'));
+    const response = await axios.get(`${process.env.API_BASE_URL}/articles?size=12`);
+    const { data: { data } } = response;
+    if (response.status === 404) {
+      dispatch(articleAction(ACTIONS.GET_ARTICLES_FAILURE, 'Article not found'));
+      dispatch(articleAction(ACTIONS.STOP_LOADING, 'stopped'));
+    }
+    dispatch(articleAction(ACTIONS.GET_ARTICLES_SUCCESS, data));
+    dispatch(articleAction(ACTIONS.STOP_LOADING, 'stopped'));
+  } catch (err) {
+    dispatch(articleAction(ACTIONS.STOP_LOADING, 'stopped'));
+    return err;
```

Comment:
#### Nitpick
- Why the return statement?

Reply:
I'll attend to that now